### PR TITLE
Depth Anything V2 - UDAV2 depth Preprocessor

### PR DESCRIPTION
**Pull request by MackinationsAi**
I've made the necessary updates for the depth_anything_v2 preprocessor to function properly (I've done multiple tests & it's working great!). After this I will be uploading the new depth_anything_v2 hf transformers.

### CODE REVIEW

**Comment by MackinationsAi** [comment]
For the time being, please use this transformers .whl that I put together for testing - https://github.com/MackinationsAi/sd-webui-controlnet/releases/tag/transformers-4.44.1.dev0. (I'm still working on getting the transformers uploaded)

**Comment by sdbds** [comment]
image
Looks like no problem, now we just need to wait for the transformer to be adapted

**Comment by kagevazquez** [comment]
if you set resize_target to True and adjust the height and width to 1036 in depth_anything_v2.py you get even better results.
Untitled
Untitled1

File: annotator/depth_anything_v2.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                encoder="vitl",
+                features=256,
+                out_channels=[256, 512, 1024, 1024],
+            )
+            .to(device)
+            .eval()
+        )
+        remote_url = os.environ.get(
+            "CONTROLNET_DEPTH_ANYTHING_V2_MODEL_URL",
+            "https://huggingface.co/MackinationsAi/Depth-Anything-V2_Safetensors/blob/main/depth_anything_v2_vitl.safetensors",
```
Comment (huchenlei) [inline annotator/depth_anything_v2.py:48]: "blob" => "resolve"
Otherwise you download an HTML file.

**Review by huchenlei — CHANGES_REQUESTED** [review]

**Comment by huchenlei** [comment]
Continued in https://github.com/Mikubill/sd-webui-controlnet/pull/2977. I don't have the push access to your branch @MackinationsAi

**Comment by zthrx** [comment]
Hi, 

How to solve this problem? 

SUPER SD 2.0 Dependencies\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\depth_anything_v2.py", line 9, in <module>
    from depth_anything_v2.dpt import DepthAnythingV2
ModuleNotFoundError: No module named 'depth_anything_v2.dpt'


Cheers!

**Comment by MackinationsAi** [comment]
@zthrx it looks like the depth_anything_v2.dpt module which is installed from the depth_anything_v2-2024.7.1.0-py2.py3-none-any.whl didn't properly install on your end. I just removed my sd-webui-controlnet extension & reinstalled it to see if the same error would happen for me which it didn't. So my advice would be to remove & reinstall sd-webui-controlnet. You should be good after that. If the issue persists open up an [issue ticket](https://github.com/Mikubill/sd-webui-controlnet/issues/new) & @ tag me so I can assist you further.

**Comment by zthrx** [comment]
> @zthrx it looks like the depth_anything_v2.dpt module which is installed from the depth_anything_v2-2024.7.1.0-py2.py3-none-any.whl didn't properly install on your end. I just removed my sd-webui-controlnet extension & reinstalled it to see if the same error would happen for me which it didn't. So my advice would be to remove & reinstall sd-webui-controlnet. You should be good after that. If the issue persists open up an [issue ticket](https://github.com/Mikubill/sd-webui-controlnet/issues/new) & @ tag me so I can assist you further.

Hey, 
Thank you for fast reply! I have a noob question, how to properly remove/reinstall this extension since I have 50gb of models. Should I remove entire folder only? 

Cheers!

**Comment by MackinationsAi** [comment]
@zthrx move the models folder somewhere else on your machine, delete the sd-webui-control folder in extensions. (Just to make sure & to be clear, I am **NOT** saying to delete your controlnet models folder in the main models folder. If your models are here - [move this folder](https://github.com/Mikubill/sd-webui-controlnet/tree/main/models) out of the sd-webui-controlnet extensions folder before deleting it). Then its a simple matter of opening up a1111 & reinstalling sd-webui-controlnet from the extensions tab & restart a1111, then move all of your models back into that folder I linked above.

**Comment by zthrx** [comment]
> @zthrx move the models folder somewhere else on your machine, delete the sd-webui-control folder in extensions. (Just to make sure & to be clear, I am **NOT** saying to delete your controlnet models folder in the main models folder. If your models are here - [move this folder](https://github.com/Mikubill/sd-webui-controlnet/tree/main/models) out of the sd-webui-controlnet extensions folder before deleting it). Then its a simple matter of opening up a1111 & reinstalling sd-webui-controlnet from the extensions tab & restart a1111, then move all of your models back into that folder I linked above.

Hey, 

Thank you for your time by helping me. Unfortunately after reinstalling I keep getting this error. Other preprocessors like Depth_anything or midas works fine. 

I've started new topic: 
https://github.com/Mikubill/sd-webui-controlnet/issues/3018

Cheers!

**Pull request closed without merging**